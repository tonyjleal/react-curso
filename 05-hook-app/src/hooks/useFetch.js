import React, { useEffect, useState } from 'react'

export const useFetch = (url) => {

    const [state, setState] = useState({ data:null, loading: true, error: null });
    
    useEffect( () => {

        setState({data:null, loading: true, error: null})

        fetch(url)
            .then( res => res.json() )
            .then( data => {
                setState({
                    data,
                    loading: false,
                    error: null,
                });
            });
    },[url]);

    return state;

    //const `https://www.breakingbadapi.com/api/quotes/${id}`


}
