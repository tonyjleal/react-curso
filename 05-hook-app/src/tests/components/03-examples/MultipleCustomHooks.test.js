import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks"
import {shallow} from 'enzyme';
import { useFetch } from "../../../hooks/useFetch";
import { useCounterMultiple } from "../../../hooks/useCounterMultiple";

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounterMultiple');


describe('Test <MultipleCustomHooks />', () => { 

    beforeEach( () => {
        useCounterMultiple.mockReturnValue({
            counter: 10,
            increment: () => {}
        })
    })

    test('should show component success', () => { 

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })
        const wrapper = shallow(<MultipleCustomHooks />);

        expect(wrapper).toMatchSnapshot();

     })

    test('should show information', () => { 
         
        useFetch.mockReturnValue({
            data: [{
                author: 'Fernando',
                quote: 'Hola mundo'
            }],
            loadin: false,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />);
        
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('Hola mundo');
        expect(wrapper.find('footer').text().trim()).toBe('Fernando');

    })



 })