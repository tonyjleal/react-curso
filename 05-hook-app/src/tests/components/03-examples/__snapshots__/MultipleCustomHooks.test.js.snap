// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <MultipleCustomHooks /> should show component success 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
